### DISCUSSION TITLE
[NEW STANDARD] - PATCH HTTP Verb should NOT be "NOT recommended"

### DISCUSSION THREAD

**Discussion by ChrisSquats** [tewhatuora/api-standards#116] [Ideas]
### Summary

Apologies for the double negative. PATCH method is a reasonable verb and a useful one for partially updating a resource, instead of replacing the resource with PUT. 

### Drawbacks

Supporting another http verb could be seen as adding complexity, but it's common practice so I don't see that as a significant drawback.

### Which area of the standards does this apply to?

- [ ] Part A - API Concepts
- [ ] Part B - API Security
- [X] Part C - API Design and Development
- [ ] Part D - FHIR
- [ ] Community guidelines

**Comment by ChrisSquats** [comment]
I'll write a proposed standard and raise a PR.

**Reply by stueynz** [reply]
The functional difference being:

1.  PUT - presents the entire Resource as you would like it to be; and is synonymous with SQL UPDATE and providing all the fields.
2. PATCH - presents a subset of fields from the Resource and their new values; and is synonymous with SQL UPDATE and providing only some fields.

The mime-types that are accepted by PATCH verb are left as a decision to the API Designer.
